delta-lake/common/src/main/scala/com/nvidia/spark/rapids/delta/GpuDeltaParquetFileFormatUtils.scala

```diff
@@ -98,29 +102,29 @@ object GpuDeltaParquetFileFormatUtils {
     closeOnExcept(rowIdxCol) { rowIdxCol =>
 
       val delVecCol = delVec.map { delVec =>
-        withResource(Scalar.fromBool(false)) { s =>
-          withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
-            var table = new Table(c)
-            val posIter = new RoaringBitmapIterator(
-              delVec.getLongIteratorFrom(rowIdxStart),
-              rowIdxStart,
-              rowIdxStart + batch.numRows(),
-            ).grouped(Math.min(maxBatchSize, batch.numRows()))
-
-            for (posChunk <- posIter) {
-              withResource(CudfColumnVector.fromLongs(posChunk: _*)) { poses =>
-                withResource(Scalar.fromBool(true)) { s =>
-                  table = withResource(table) { _ =>
-                    Table.scatter(Array(s), poses, table)
-                  }
-                }
+        delVectorScatterTimeMetric.ns {
+          withResource(Scalar.fromBool(false)) { s =>
+            withResource(CudfColumnVector.fromScalar(s, batch.numRows())) { c =>
```

Comment:
`c` is not needed after the first table is built.  Keeping it allocated throughout means we effectively double the GPU memory utilization of the algorithm.

Reply:
Similarly the `false` form of `s` is not needed after `c` is built.  The scalar is tiny from a memory utilization perspective, but it would be good to clean it up and avoid the shadowing of `s` which occurs with the opposite value scalar below.